b2b_ecommerce/models_test.py

```diff
@@ -52,3 +55,71 @@ def test_reference_number(settings):
         f"{REFERENCE_NUMBER_PREFIX}{cybersource_prefix}-{order.id}"
         == order.reference_number
     )
+
+
+def test_get_unexpired_coupon(order_with_coupon):
+    """get_unexpired_coupon should get a coupon matching the product id and coupon code, which is also valid"""
+    coupon = order_with_coupon.coupon
+    assert (
+        B2BCoupon.objects.get_unexpired_coupon(
+            coupon_code=coupon.coupon_code, product_id=coupon.product_id
+        )
+        == coupon
+    )
+
+
+@pytest.mark.parametrize(
+    "activation_date, expiration_date",
+    [
+        [None, None],
+        [timezone.now() - timedelta(days=1), timezone.now() + timedelta(days=1)],
+        [None, timezone.now() + timedelta(days=1)],
+        [timezone.now() - timedelta(days=1), None],
+    ],
+)
+def test_get_unexpired_coupon_null_dates(
```

Comment:
Nit: change to `test_get_unexpired_coupon`. This is now just basically testing the happy path for the function, not just for null dates